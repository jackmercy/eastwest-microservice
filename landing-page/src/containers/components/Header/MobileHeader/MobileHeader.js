import Link from 'next/link';
import { useRouter } from 'next/router';
import { memo, useCallback, useEffect, useMemo, useState } from 'react';
import { useIntl } from 'react-intl';
import classNames from 'classnames';

// MVP hidden
// eslint-disable-next-line no-unused-vars
import { Logo, Button } from '../../../../components';
import { useEventListener } from '../../../../hooks';

import useStyles from './mobileHeader.styles';

const MobileHeader = () => {
  const intl = useIntl();
  const classes = useStyles();
  const router = useRouter();
  const [activatedMenu, setActiveMenu] = useState(false);
  const [activeatedHeader, setActiveHeader] = useState(false);
  const products = useMemo(
    () => [
      {
        key: 'talent-labs',
        href: '/talent-labs',
        logo: 'talent-labs-black-no-text',
        name: intl.formatMessage({ id: 'HEADER.PRODUCTS_SERVICES.1.TITLE' }),
      },
      // MVP hidden
      // {
      //   key: 'art',
      //   href: '/art',
      //   logo: 'art-red-no-text',
      //   name: intl.formatMessage({ id: 'HEADER.PRODUCTS_SERVICES.2.TITLE' }),
      // },
      // {
      //   key: 'dashboard',
      //   href: '/dashboard',
      //   logo: 'dashboard-black-no-text',
      //   name: intl.formatMessage({ id: 'HEADER.PRODUCTS_SERVICES.3.TITLE' }),
      // },
      {
        key: 'data-labs',
        href: '/data-labs',
        logo: 'data-labs-red-no-text',
        name: intl.formatMessage({ id: 'HEADER.PRODUCTS_SERVICES.4.TITLE' }),
      },
    ],
    [intl],
  );
  const Products = useCallback(
    () => (
      <div className={classes.productMenu}>
        {products.map((item) => (
          <div className={classes.productItem} key={item.key}>
            <Logo
              href={item.href}
              alt={item.name}
              type={item.logo}
              className={classes.productLogo}
            />
            <Link href={item.href} passHref>
              <a href={item.href} className={classes.productName}>
                {item.name}
              </a>
            </Link>
          </div>
        ))}
      </div>
    ),
    [activatedMenu, products, classes],
  );

  useEffect(
    () => () => {
      const element = document.getElementsByTagName('body');
      if (element && activatedMenu) {
        element[0].style.overflow = !activatedMenu ? 'hidden' : null;
      }
    },
    [activatedMenu],
  );

  const handleItemMenuClick = (e) => {
    e.preventDefault();
  };

  const NavItem = useCallback(
    ({ href, label, active, dataTest, children, ...props }) => (
      <div className={classNames(classes.navItem, { active })}>
        <Link href={href} passHref>
          <a
            href={href}
            className={classes.navLink}
            data-test={dataTest}
            {...props}
          >
            {label}
          </a>
        </Link>
        {children}
      </div>
    ),
    [],
  );

  const configs = useMemo(
    () => [
      {
        key: 'products',
        component: NavItem,
        href: '/#',
        onClick: handleItemMenuClick,
        dataTest: 'header-url-products',
        label: intl.formatMessage({ id: 'HEADER.PRODUCTS_SERVICES' }),
        children: <Products />,
      },
      // MVP hidden
      // {
      //   key: 'pricing',
      //   component: NavItem,
      //   href: '/pricing',
      //   active: router.pathname === '/pricing',
      //   dataTest: 'header-url-pricing',
      //   label: intl.formatMessage({ id: 'HEADER.PRICING' }),
      // },
      {
        key: 'company',
        component: NavItem,
        href: '/company',
        active: router.pathname === '/company',
        dataTest: 'header-url-company',
        label: intl.formatMessage({ id: 'HEADER.COMPANY' }),
      },
      // MVP hidden
      // {
      //   key: 'support',
      //   component: NavItem,
      //   href: '/support',
      //   active: router.pathname === '/support',
      //   dataTest: 'header-url-support',
      //   label: intl.formatMessage({ id: 'HEADER.SUPPORT' }),
      // },
    ],
    [intl, router, classes, handleItemMenuClick],
  );

  useEventListener(
    'scroll',
    () => {
      if (!activatedMenu) {
        const top = window.pageYOffset || document.documentElement.scrollTop;
        setActiveHeader(top > 50);
      }
    },
    true,
  );

  const toggleMenu = () => {
    const element = document.getElementsByTagName('body');
    if (element) {
      element[0].style.overflow = !activatedMenu ? 'hidden' : null;
    }
    setActiveMenu(!activatedMenu);
  };

  return (
    <header
      className={classNames(classes.root, {
        active: activeatedHeader,
        showMenu: activatedMenu,
      })}
    >
      <div className={classes.main}>
        <Logo className={classes.logo} dataTest="header-logo" />
        <span
          role="presentation"
          onClick={toggleMenu}
          className={classes.toggleMenuButton}
        >
          <i className={!activatedMenu ? 'icon-hamberger' : 'icon-close'} />
        </span>
      </div>
      <div className={classNames(classes.menu, { active: activatedMenu })}>
        <div className={classNames('custom-scrollbar', classes.navbar)}>
          {configs.map(({ key, component: Component, ...restProps }) => (
            <Component key={key} {...restProps} />
          ))}
        </div>
        {/* MVP hidden */}
        {/* <div className={classes.footer}>
          <Button
            size="large"
            color="tertiary"
            className={classes.button}
            keepSizeOnMobile
            dataTest="header-url-sign-in"
          >
            {intl.formatMessage({ id: 'HEADER.SIGN_IN' })}
          </Button>
          <Button
            size="large"
            color="secondary"
            className={classes.button}
            keepSizeOnMobile
            dataTest="header-url-sign-up"
          >
            {intl.formatMessage({ id: 'HEADER.GET_STARTED' })}
          </Button>
        </div> */}
      </div>
    </header>
  );
};

export default memo(MobileHeader);
