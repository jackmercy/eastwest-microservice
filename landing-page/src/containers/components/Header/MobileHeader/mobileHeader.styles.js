import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    top: 0,
    zIndex: 99,
    width: '100%',
    position: 'fixed',
    boxSizing: 'border-box',
    backgroundColor: COLORS.bgWhite,
    padding: [0, rem(16)],
    transition: 'padding .3s ease, min-height .5s ease',
    display: 'flex',
    flexDirection: 'column',
    minHeight: 0,
    '&.active': { boxShadow: '0px 4px 20px rgba(0, 0, 0, 0.15)' },
    '&.showMenu': {
      minHeight: '100vh',
      height: '100%',
      overflow: 'hidden',
    },
  },
  main: {
    zIndex: 2,
    flex: 'none',
    width: '100%',
    display: 'flex',
    padding: [rem(16), 0],
    justifyContent: 'space-between',
    position: 'relative',
    alignItems: 'center',
    boxSizing: 'border-box',
    backgroundColor: 'transparent',
  },
  logo: {
    flex: 'none',
    marginRight: rem(80),
    maxWidth: rem(206),
    transition: 'margin .3s ease, max-width .5s ease',
  },
  toggleMenuButton: { fontSize: rem(28), cursor: 'pointer' },
  menu: {
    maxHeight: 0,
    display: 'flex',
    flexDirection: 'column',
    flex: '1 1 auto',
    margin: [0, rem(-16)],
    boxSizing: 'border-box',
    backgroundColor: 'transparent',
    overflow: 'hidden',
    transition: 'max-height .5s ease',
    '&.active': { maxHeight: '100vh' },
  },
  navbar: {
    padding: 0,
    height: '100%',
    flex: '1 1 auto',
    overflowX: 'hidden',
    overflowY: 'auto',
    paddingBottom: rem(50),
    boxSizing: 'border-box',
    backgroundColor: 'transparent',
    borderTop: `1px solid ${COLORS.borderGray01}`,
  },
  navItem: {
    flex: 'none',
    padding: [0, rem(16)],
    borderBottom: `1px solid ${COLORS.borderGray01}`,
    '&.active': { borderBottom: `${rem(2)} solid ${COLORS.primary}` },
  },
  navLink: {
    padding: [rem(16), 0],
    display: 'block',
    textDecoration: 'none',
    color: COLORS.textBlack01,
    fontWeight: 600,
    fontSize: rem(16),
    lineHeight: rem(24),
  },
  footer: {
    flex: 'none',
    padding: [rem(16), rem(32)],
    textAlign: 'right',
    margin: [0, rem(-16)],
    backgroundColor: COLORS.bgGray02,
  },
  button: {
    '&.tertiary': {
      color: COLORS.primary,
      marginRight: rem(16),
    },
    '&.secondary': { backgroundColor: 'transparent' },
  },
  productMenu: { paddingBottom: rem(24) },
  productItem: {
    display: 'flex',
    alignItems: 'center',
    boxSizing: 'border-box',
    '& + $productItem': { marginTop: rem(32) },
  },
  productLogo: {
    flex: 'none',
    width: rem(48),
    height: rem(48),
    marginRight: rem(32),
  },
  productName: {
    margin: 0,
    fontWeight: 900,
    fontSize: rem(32),
    lineHeight: rem(48),
    textDecoration: 'none',
    color: COLORS.textBlack01,
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    logo: { marginRight: rem(40), maxWidth: rem(126) },
    toggleMenuButton: { fontSize: rem(16) },
  },
});
