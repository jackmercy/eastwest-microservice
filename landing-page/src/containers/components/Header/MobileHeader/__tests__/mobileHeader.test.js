import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MobileHeader from '../MobileHeader';

describe('Testing Header > MobileHeader', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<MobileHeader />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<MobileHeader />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a logo', () => {
    expect(wrapper.find('[data-test="header-logo"]')).toHaveLength(1);
  });

  it('should have a services dropdown introduce', () => {
    expect(wrapper.find('[data-test="header-url-products"]')).toHaveLength(1);
  });

  // MVP hidden
  // it('should have a pricing url', () => {
  //   expect(wrapper.find('[data-test="header-url-pricing"]')).toHaveLength(1);
  // });

  it('should have a compnay url', () => {
    expect(wrapper.find('[data-test="header-url-company"]')).toHaveLength(1);
  });

  // MVP hidden
  // it('should have a support url', () => {
  //   expect(wrapper.find('[data-test="header-url-support"]')).toHaveLength(1);
  // });

  // it('should have a sign in url', () => {
  //   expect(wrapper.find('[data-test="header-url-sign-in"]')).toHaveLength(1);
  // });

  // it('should have a sign up url', () => {
  //   expect(wrapper.find('[data-test="header-url-sign-up"]')).toHaveLength(1);
  // });
});
