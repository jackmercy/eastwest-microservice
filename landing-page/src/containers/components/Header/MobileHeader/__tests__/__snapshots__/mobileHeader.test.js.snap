// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header > MobileHeader should render match snapshot 1`] = `
<header
  className="root-0-2-1"
>
  <div
    className="main-0-2-2"
  >
    <Memo(Logo)
      className="logo-0-2-3"
      dataTest="header-logo"
    />
    <span
      className="toggleMenuButton-0-2-4"
      onClick={[Function]}
      role="presentation"
    >
      <i
        className="icon-hamberger"
      />
    </span>
  </div>
  <div
    className="menu-0-2-5"
  >
    <div
      className="custom-scrollbar navbar-0-2-6"
    >
      <Component
        dataTest="header-url-products"
        href="/#"
        key="products"
        label="Products & Services"
        onClick={[Function]}
      >
        <Component />
      </Component>
      <Component
        active={false}
        dataTest="header-url-company"
        href="/company"
        key="company"
        label="Company"
      />
    </div>
  </div>
</header>
`;
